src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1189,15 +1191,23 @@ CopyLocalDataIntoShards(Oid distributedRelationId)
 	slot = MakeSingleTupleTableSlot(tupleDescriptor);
 	columnNameList = TupleDescColumnNameList(tupleDescriptor);
 
+	/* determine the partition column */
```

Comment:
This might be a bit confusing to the readers (at first I was surprised as well thinking about `COPY table (col5, col2)` type of syntax where the user could easily change the order of the columns).

So, we're (and should be) doing this because Postgres adds dropped columns to the `slot->tts_values` anyway, right?  (Some debugging verified that for me)

It'd be nice to add a comment for that.